src/AltinnCore/Common/Services/Implementation/AuthenticationAppSI.cs

```diff
@@ -0,0 +1,68 @@
+using System;
+using System.IO;
+using System.Net;
+using System.Net.Http;
+using System.Threading.Tasks;
+using System.Xml.Serialization;
+using AltinnCore.Authentication.JwtCookie;
+using AltinnCore.Authentication.Utils;
+using AltinnCore.Common.Clients;
+using AltinnCore.Common.Configuration;
+using AltinnCore.Common.Helpers;
+using AltinnCore.Common.Services.Interfaces;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace AltinnCore.Common.Services.Implementation
+{
+    /// <summary>
+    /// Implementation for authentication service
+    /// </summary>
+    public class AuthenticationAppSI : IAuthentication
+    {
+        private readonly ILogger _logger;
+        private readonly PlatformSettings _platformSettings;
+        private readonly IHttpContextAccessor _httpContextAccessor;
+        private readonly JwtCookieOptions _cookieOptions;
+        private readonly HttpClient _client;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ProfileAppSI"/> class
+        /// </summary>
+        /// <param name="logger">the logger</param>
+        /// <param name="platformSettings">the platform settings</param>
+        /// <param name="httpContextAccessor">The http context accessor </param>
+        /// <param name="cookieOptions">The cookie options </param>
+        /// <param name="httpClientAccessor">The http client accessor </param>
+        public AuthenticationAppSI(
+            ILogger<AuthenticationAppSI> logger,
+            IOptions<PlatformSettings> platformSettings,
+            IHttpContextAccessor httpContextAccessor,
+            IOptions<JwtCookieOptions> cookieOptions,
+            IHttpClientAccessor httpClientAccessor)
+        {
+            _logger = logger;
+            _platformSettings = platformSettings.Value;
+            _httpContextAccessor = httpContextAccessor;
+            _cookieOptions = cookieOptions.Value;
+            _client = httpClientAccessor.AuthenticationClient;
+        }
+
+        /// <inheritdoc />
+        public async Task<HttpStatusCode> RefreshToken()
+        {
+            string endpointUrl = $"refresh";
+            string token = JwtTokenUtil.GetTokenFromContext(_httpContextAccessor.HttpContext, _cookieOptions.Cookie.Name);
+            JwtTokenUtil.AddTokenToRequestHeader(_client, token);
+            HttpResponseMessage response = await _client.GetAsync(endpointUrl);
+            if (response.StatusCode != System.Net.HttpStatusCode.OK)
+            {
+                _logger.LogError($"Refreshing JwtToken failed with status code {response.StatusCode}");
+            }
+
+            return response.StatusCode;
```

Comment:
Dont we need to send the updated cookie back to client?

Reply:
When signing back into the context the cookie is set to the at21.altinn.cloud domain, should that be sufficient or should we explicitly return the cookie too?